Historical Pegin Validation

scottmillner opened the pull request:
… Factory

scottmillner committed:
refactor: add botanix_consensus_package to HeaderExt, create Bitcoind Factory

arminsabouri commented on crates/primitives/src/header_ext.rs:
```diff
@@ -161,17 +171,37 @@ impl PartialEq for ValidateAuthoritySignatureError {
     fn eq(&self, other: &Self) -> bool {
         matches!(
             (self, other),
-            (Self::InvalidAuthority, Self::InvalidAuthority) |
-                (Self::InvalidMessage, Self::InvalidMessage) |
-                (Self::InvalidSignature, Self::InvalidSignature) |
-                (Self::MissingSignature, Self::MissingSignature) |
-                (Self::InvalidSignerIndex(_), Self::InvalidSignerIndex(_)) |
-                (Self::RecoverFailed, Self::RecoverFailed) |
-                (Self::InvalidEdhFormat(_), Self::InvalidEdhFormat(_))
+            (Self::InvalidAuthority, Self::InvalidAuthority)
+                | (Self::InvalidMessage, Self::InvalidMessage)
+                | (Self::InvalidSignature, Self::InvalidSignature)
+                | (Self::MissingSignature, Self::MissingSignature)
+                | (Self::InvalidSignerIndex(_), Self::InvalidSignerIndex(_))
+                | (Self::RecoverFailed, Self::RecoverFailed)
+                | (Self::InvalidEdhFormat(_), Self::InvalidEdhFormat(_))
```
rustfmt has no chill

arminsabouri reviewed (dismissed):
Approach Ack

arminsabouri committed:
refactor(BTE): use bitcoind factory in tree externals

arminsabouri committed:
refactor(header_ext): use bitcoindFactory trait impl

arminsabouri committed:
chore: remove unused `broadcast_tx` function from bitcoind interface

arminsabouri committed:
chore(btc_wallet): remove unused block source interface

arminsabouri committed:
chore: expose bitcoind json rpc lib to workspace

arminsabouri committed:
refactor: bitcoind factory should return `impl RpcApi` instead a specific client

arminsabouri committed:
refactor(BTE): externals doesnt need bitcoind factory

arminsabouri committed:
refactor: botanix consensus pkg should be created in `EVMProcessor.transact`

scottmillner committed:
refactor: start removing botanix_consensus_package where not needed

arminsabouri committed:
configure executor factory with bitcoind L1 resources

block builder and fetcher don't use botanix_consensus_pkg, create RpcApiExt for bitcoind, remove remaining botanix_consensus_pkg where not needed

arminsabouri committed:
WIP - create EVMProcessor with bitcoind factory in block fetcher and block builder

scottmillner committed:
refactor: initialize pipeline bitcoind factory and network

arminsabouri committed:
chore: clean up warnings + unused crate dependencies

arminsabouri committed:
chore(evm): logging block builder address should be debug

scottmillner committed:
test: add test for botanix_consensus_package()

test: add assertions

scottmillner committed:
test: add MockBitcoindFactory to blockchain_test so ethereum tests pass, add MockBitcoindFactory where needed

scottmillner committed:
refactor: rpc node should set aggregate_public_key to None in block_fetcher

arminsabouri committed:
fix(op): processor BF trait def
